src/main/java/org/openhab/automation/jrule/internal/JRuleUtil.java

```diff
@@ -90,13 +92,19 @@ public static <T> CompletableFuture<T> delayedExecution(long delay, TimeUnit uni
     }
 
     public static <T> CompletableFuture<T> scheduleAsync(Supplier<CompletableFuture<T>> command, long delay,
-            TimeUnit unit) {
+            TimeUnit unit, JRuleExecutionContext context) {
         CompletableFuture<T> completableFuture = new CompletableFuture<>();
         scheduler.schedule((() -> {
-            command.get().thenAccept(completableFuture::complete).exceptionally(t -> {
-                completableFuture.completeExceptionally(t);
-                return null;
-            });
+            try {
+                JRule.JRULE_EXECUTION_CONTEXT.set(context);
```

Comment:
its hard to get a full overview inside the webui, but isn't it possible without using this static field var?

Reply:
I think it is a good candidate for being static - as we only need one?

Reply:
all right, just asking. for thread-local it make sense to be static, but its used in multiple classes.
when you say its fine, its fine.